src/main/scala/workflow/graph/PipelineEnv.scala

```diff
@@ -30,6 +30,14 @@ class PipelineEnv {
   def setOptimizer(optimizer: Optimizer): Unit = {
     _optimizer = optimizer
   }
+
+  /**
+   * Reset this PipelineEnv (clear state and set the Optimizer to the DefaultOptimizer)
+   */
+  def reset(): Unit = {
+    state.clear()
+    setOptimizer(DefaultOptimizer)
```

Comment:
Would a user want this functionality or does this belong in PipelineContext?

On Jun 3, 2016, at 9:02 AM, Tomer Kaftan <notifications@github.com> wrote:

In src/main/scala/workflow/graph/PipelineEnv.scala
<https://github.com/amplab/keystone/pull/279#discussion_r65732447>:

> @@ -30,6 +30,14 @@ class PipelineEnv {
>    def setOptimizer(optimizer: Optimizer): Unit = {
>      _optimizer = optimizer
>    }
> +
> +  /**
> +   * Reset this PipelineEnv (clear state and set the Optimizer to the DefaultOptimizer)
> +   */
> +  def reset(): Unit = {
> +    state.clear()
> +    setOptimizer(DefaultOptimizer)

It resets the Optimizer to the default. This is because some unit tests
change the optimizer to test different rules or validate things about the
execution, and we eventually have to change it back for other unit tests.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/amplab/keystone/pull/279/files/653be30f2fdd1eb649cc69cb23929728c3d74c42#r65732447>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABQ8Ze3p_86Uu21nev8Wa_LbgoBqUS7tks5qIFABgaJpZM4ItH40>
.